flink-ml-core/src/test/java/org/apache/flink/ml/linalg/SparseVectorTest.java

```diff
@@ -47,6 +49,16 @@ public void testConstructor() {
         assertArrayEquals(indices, vector.indices);
         assertArrayEquals(values, vector.values, 1e-5);
         assertEquals("(4, [0, 2, 3], [0.1, 0.3, 0.4])", vector.toString());
+
+        Map<Integer, Double> map = new HashMap<>(4);
+        map.put(0, 0.1);
+        map.put(2, 0.3);
+        map.put(3, 0.4);
+        SparseVector mapVector = Vectors.sparse(n, map);
+        assertEquals(n, mapVector.n);
+        assertArrayEquals(indices, mapVector.indices);
+        assertArrayEquals(values, mapVector.values, 1e-5);
+        assertEquals("(4, [0, 2, 3], [0.1, 0.3, 0.4])", mapVector.toString());
```

Comment:
Can we also add test cases for the following situations?
- Indices are invalid. e.g., indices are negative or larger than the size of the vector.
- Indices contain duplicated values.
- Some values are 0.

Reply:
In map constructor, indices are keys in map. There is no possible existing duplicated indices.

1 and 3  has refined.